src/diart/websockets.py

```diff
@@ -0,0 +1,298 @@
+
+            # Setup RTTM response hook
+            self._clients[client_id].inference.attach_hooks(
+                lambda ann_wav: self.send(client_id, ann_wav[0].to_rttm())
+            )
+
+            # Start inference
+            self._clients[client_id].inference()
+            logger.info(f"Started inference for client: {client_id}")
+
+            # Send ready notification to client
+            self.send(client_id, "READY")
+        except Exception as e:
+            logger.error(f"Failed to initialize client {client_id}: {e}")
+            self.close(client_id)
+
+    def _on_disconnect(self, client: Dict[Text, Any], server: WebsocketServer) -> None:
+        """Handle client disconnection.
+
+        Parameters
+        ----------
+        client : Dict[Text, Any]
+            Client information dictionary
+        server : WebsocketServer
+            WebSocket server instance
+        """
+        client_id = client["id"]
+        logger.info(f"Client disconnected: {client_id}")
+        self.close(client_id)
+
+    def _on_message_received(
+        self, client: Dict[Text, Any], server: WebsocketServer, message: AnyStr
+    ) -> None:
+        """Process incoming client messages.
+
+        Parameters
+        ----------
+        client : Dict[Text, Any]
+            Client information dictionary
+        server : WebsocketServer
+            WebSocket server instance
+        message : AnyStr
+            Received message data
+        """
+        client_id = client["id"]
+
+        if client_id not in self._clients:
+            return
+
+        try:
+            self._clients[client_id].audio_source.process_message(message)
+        except (socket.error, ConnectionError) as e:
+            logger.warning(f"Client {client_id} disconnected: {e}")
+            self.close(client_id)
+        except Exception as e:
+            logger.error(f"Error processing message from client {client_id}: {e}")
+            # Don't close the connection for non-connection related errors
+            # This allows the client to retry sending the message
+
+    def send(self, client_id: Text, message: AnyStr) -> None:
+        """Send a message to a specific client.
+
+        Parameters
+        ----------
+        client_id : Text
+            Target client identifier
+        message : AnyStr
+            Message to send
+        """
+        if not message:
+            return
+
+        client = next((c for c in self.server.clients if c["id"] == client_id), None)
+
+        if client is None:
+            return
+
+        try:
+            self.server.send_message(client, message)
+        except (socket.error, ConnectionError) as e:
+            logger.warning(
+                f"Client {client_id} disconnected while sending message: {e}"
+            )
+            self.close(client_id)
+        except Exception as e:
+            logger.error(f"Failed to send message to client {client_id}: {e}")
```

Comment:
Should we handle retries in case of predictions failing to send? Both make sense in a way because you might want fault tolerance, but you also might want only the latest results, no matter if you lose some data in the middle. Maybe we could make this configurable when instantiating the server. Something like:
```python
server = WebSocketStreamingServer(..., retry_delivery=True, max_delivery_retries=3)
```

Concerning the implementation, we could have a message delivery queue (possibly per client) and consume it in a separate thread or coroutine